### ISSUE TITLE
GTK Integration Issue

### ISSUE THREAD

**Issue by Cybercountry** [flathub/io.ente.auth#3]
The application does not seem to adhere to the system's GTK theme, resulting in a non-native appearance.

**Error Messages:**

Upon launching the application, I encountered the following messages in the terminal:

```
Gtk-Message: 09:55:52.090: Failed to load module "xapp-gtk3-module"
flutter: [ente_logging] [INFO] [2024-10-07 09:55:52.353624] log file for today: File: '/home/[user]/.var/app/io.ente.auth/data/enteauth/logs/2024-10-7.txt' with prefix 
flutter: [ente_logging] [INFO] [2024-10-07 09:55:52.353624] app version: '4.0.3+403'
flutter: [ente_logging] [INFO] [2024-10-07 09:55:52.353624] 
flutter: [main] [INFO] [2024-10-07 09:55:52.353685] Starting app in foreground
flutter: [Configuration] [INFO] [2024-10-07 09:55:52.353770] Skipping temp folder clear
(enteauth:2): Gdk-WARNING **: 09:55:52.562: Native Windows wider or taller than 32767 pixels are not supported
flutter: Building OnboardingPage
flutter: [UpdateService] [INFO] [2024-10-07 09:55:52.742655] Debouncing notification
```

**Screenshot:**
![image](https://github.com/user-attachments/assets/1c9476dd-7e86-4b6e-b0ad-f3fce959201d)

**Environment:**

- **Operating System:** Arch Linux 6.6.52-1-lts
- **Desktop Environment:** Gnome 47

Thanks to those responsible for this Flatpak version of Ente Auth, it's a very grateful contribution.

**Comment by prateekmedia** [comment]
This seems more like a flatpak issue more than an ente auth issue, correct me if I am wrong.

**Comment by prateekmedia** [comment]
See:
https://gist.github.com/lmintmate/b512680c0d1ee8f41b8f43cd2c81dc2c

**Comment by Cybercountry** [comment]
> This seems more like a general flatpak issue more than an ente auth issue, correct me if I am wrong.

I believe this could be true, since the same does not occur in the appimage version. However, I'm not sure if it could not be something related to flutter.

**Comment by prateekmedia** [comment]
Could you try some other gtk app flatpak and verify.

**Comment by Cybercountry** [comment]
> Could you try some other gtk app flatpak and verify.



> Could you try some other gtk app flatpak and verify.

Among my applications, Ente Auth was the first one where this happened. The other applications, mainly after Gnome 47, adopt this bar:
![image](https://github.com/user-attachments/assets/ae8ff26e-8c51-4d8b-b216-bd0de09589b9)

**Comment by Cybercountry** [comment]
> > Could you try some other gtk app flatpak and verify.
> 
> Among my applications, Ente Auth was the first one where this happened. The other applications, mainly after Gnome 47, adopt this bar: ![image](https://private-user-images.githubusercontent.com/146500015/374217105-ae8ff26e-8c51-4d8b-b216-bd0de09589b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzMTI1MjQsIm5iZiI6MTcyODMxMjIyNCwicGF0aCI6Ii8xNDY1MDAwMTUvMzc0MjE3MTA1LWFlOGZmMjZlLThjNTEtNGQ4Yi1iMjE2LWJkMGRlMDk1ODliOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwN1QxNDQzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZWZlZThkOGFjZmRkZmM4ODAxYTNiZTA2ZDFiZWJkYjMyYTIwZTI4ZGYyMzMxMGI3MmE2ZTAzZWE4MmRkMzUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Lg6qT2dSSspdMyeDCMlv_6rzqU7dqaamdFVWgZxDAs)

Another thing that could be related to dark/light mode. Ente Auth may not be recognizing my system's dark theme and displaying this bar.

**Comment by prateekmedia** [comment]
Ente Auth is a gtk 3 application, so might wanna see similar gtk 3 applications.

**Comment by Cybercountry** [comment]
> Ente Auth is a gtk 3 application, so might wanna see similar gtk 3 applications.

I may be wrong, but I believe that signal is gtk3 too. But anyway, one that I'm sure is gtk3 is the email client I use, Geary:
![image](https://github.com/user-attachments/assets/4d756532-5c53-41ca-9cb7-a277d76753e7)

And GIMP:
![image](https://github.com/user-attachments/assets/61706d97-92a8-486d-ac08-0e062ef11448)

**Comment by Cybercountry** [comment]
> > > Could you try some other gtk app flatpak and verify.
> > 
> > 
> > Among my applications, Ente Auth was the first one where this happened. The other applications, mainly after Gnome 47, adopt this bar: ![image](https://private-user-images.githubusercontent.com/146500015/374217105-ae8ff26e-8c51-4d8b-b216-bd0de09589b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzMTI1MjQsIm5iZiI6MTcyODMxMjIyNCwicGF0aCI6Ii8xNDY1MDAwMTUvMzc0MjE3MTA1LWFlOGZmMjZlLThjNTEtNGQ4Yi1iMjE2LWJkMGRlMDk1ODliOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwN1QxNDQzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZWZlZThkOGFjZmRkZmM4ODAxYTNiZTA2ZDFiZWJkYjMyYTIwZTI4ZGYyMzMxMGI3MmE2ZTAzZWE4MmRkMzUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8Lg6qT2dSSspdMyeDCMlv_6rzqU7dqaamdFVWgZxDAs)
> 
> Another thing that could be related to dark/light mode. Ente Auth may not be recognizing my system's dark theme and displaying this bar.


Since I mentioned Geary, I remember that when I first installed it, it had the same issue: the top bar didn’t recognize the dark theme from my system. I looked through the [project's repository](https://gitlab.gnome.org/GNOME/geary) issues extensively until I was able to fix it. Unfortunately, I don’t remember the exact steps I took, but I believe I installed something like org.gtk.Gtk3theme.adw-gtk3-dark and replace something from Geary. However, this was last year, so I’m not entirely sure of the process. I'll attach a screenshot showing how it looked back then.
![image](https://github.com/user-attachments/assets/a2fdbbb5-3c9b-4aa4-b082-6a236eb0a0c5)

**Comment by prateekmedia** [comment]
You're right that was the fix. But since you are saying it didn't fixed for you, maybe you can try to set the theme to Dark theme from the side drawer although it may not be adapted by header.

**Comment by Cybercountry** [comment]
> You're right that was the fix. But since you are saying it didn't fixed for you, maybe you can try to set the theme to Dark theme from the side drawer although it may not be adapted by header.

I tried opening Ente Auth with the following command:
`flatpak run --env=GTK_THEME=adw-gtk3-dark io.ente.auth`

![image](https://github.com/user-attachments/assets/5aa46023-a04a-41ae-b3bd-db352cdbac2a)

As you can see in the screenshot I attached, the application opened correctly with the bar recognizing the dark theme of my system. However, this setting is not saved, that is, after closing the application and launching it again from the desktop shortcut, it returns to the previous state.

Do you know if there is any way to make this setting permanent?

**Comment by prateekmedia** [comment]
Could you tweak the theme from gnome tweaks?

**Comment by prateekmedia** [comment]
Maybe changing gnome tweaks to dark theme would fix this.

**Comment by prateekmedia** [comment]
Also if you know Flatseal then you can manually add this environment variable to ente auth too.

**Comment by Cybercountry** [comment]
> Also if you know Flatseal then you can manually add this environment variable to ente auth too.

Done!
I set the following environment variable through Flatseal: `GTK_THEME=adw-gtk3-dark`
And now it is working properly with dark mode!
Thank you very much for your attention and support!
